src/components/app/details/cicdHistory/LogsRenderer.tsx

```diff
@@ -0,0 +1,159 @@
+        renderConfigurationError(isBlobStorageConfigured)
+    ) : (
+        <div className="logs__body">
+            {logs.map((log, index) => {
+                return <p className="mono fs-14" key={`logs-${index}`} dangerouslySetInnerHTML={createMarkup(log)} />
+            })}
+            {(triggerDetails.podStatus === POD_STATUS.PENDING || (eventSource && eventSource.readyState <= 1)) && (
+                <div className="flex left event-source-status">
+                    <Progressing />
+                </div>
+            )}
+        </div>
+    )
+}
+
+function useCIEventSource(url: string, maxLength?: number) {
+    const [data, setData] = useState([])
+    let retryCount = LOGS_RETRY_COUNT
+    const [logsNotAvailableError, setLogsNotAvailableError] = useState<boolean>(false)
+    const [interval, setInterval] = useState(1000)
+    const buffer = useRef([])
+    const eventSourceRef = useRef(null)
+    useInterval(populateData, interval)
+
+    function populateData() {
+        setData((data) => [...data, ...buffer.current])
+        buffer.current = []
+    }
+    function closeEventSource() {
+        if (eventSourceRef.current && eventSourceRef.current.close) eventSourceRef.current.close()
+    }
+
+    function handleMessage(event) {
+        if (event.type === 'message') {
+            retryCount = LOGS_RETRY_COUNT
+            buffer.current.push(event.data)
+        }
+    }
+
+    function handleStreamStart() {
+        retryCount = LOGS_RETRY_COUNT
+        buffer.current = []
+        setData([])
+    }
+
+    function handleStreamEnd() {
+        retryCount = LOGS_RETRY_COUNT
+        setData((data) => [...data, ...buffer.current])
+        buffer.current = []
+        eventSourceRef.current.close()
+        setInterval(null)
+    }
+
+    function handleError(error: any) {
+        retryCount--
+        if (eventSourceRef.current) {
+            eventSourceRef.current.close()
+        }
+        if (retryCount > 0) {
+            getData()
+        } else {
+            setLogsNotAvailableError(true)
+            setInterval(null)
+        }
+    }
+
+    function getData() {
+        buffer.current = []
+        eventSourceRef.current = new EventSource(url, { withCredentials: true })
+        eventSourceRef.current.addEventListener(EVENT_STREAM_EVENTS_MAP.MESSAGE, handleMessage)
+        eventSourceRef.current.addEventListener(EVENT_STREAM_EVENTS_MAP.START_OF_STREAM, handleStreamStart)
+        eventSourceRef.current.addEventListener(EVENT_STREAM_EVENTS_MAP.END_OF_STREAM, handleStreamEnd)
+        eventSourceRef.current.addEventListener(EVENT_STREAM_EVENTS_MAP.ERROR, handleError)
+    }
+
+    useEffect(() => {
+        if (url) {
+            getData()
+        }
+        return closeEventSource
+    }, [url, maxLength])
+
+    return [data, eventSourceRef.current, logsNotAvailableError]
+}
+
+const renderLogsNotAvailable = (subtitle?: string): JSX.Element => {
```

Comment:
can we keep these inside parents as they're not used anywhere else?

Reply:
Done